crates/bitwarden/src/vault/cipher/attachment.rs

```diff
@@ -80,15 +80,20 @@ impl KeyDecryptable<SymmetricCryptoKey, Vec<u8>> for AttachmentFile {
         let ciphers_key = Cipher::get_cipher_key(key, &self.cipher.key)?;
         let ciphers_key = ciphers_key.as_ref().unwrap_or(key);
 
-        let mut attachment_key: Vec<u8> = self
+        // Decrypt the attachment key if there is one
```

Comment:
I wonder if this code would be more easily understood,

```rust
/// Version 2 or 3, `AttachmentKey` or `CipherKey(AttachmentKey)`
if let Some(attachment_key) = &self.attachment.key {
    let mut content_key: Vec<u8> = attachment_key.decrypt_with_key(ciphers_key)?;
    let content_key = SymmetricCryptoKey::try_from(content_key.as_mut_slice())?;

    self.contents.decrypt_with_key(&content_key)
} else {
    /// Legacy attachment, use user/org key
    self.contents.decrypt_with_key(key)
}
```

Reply:
Agreed, changed it in 0cbed5cbbaa01749092781d66be891a7aa6ffa02